doc/source/authentication.rst

```diff
@@ -191,3 +191,64 @@ You can specify this list of usernames in your `config.yaml`:
        users:
          - user1
          - user2
+
+Advanced: adding your own Ingress object
+----------------------------------------
+
+If you are using a Kubernetes Cluster that does not provide public IPs for
+services directly, you need to use
+an `ingress <https://kubernetes.io/docs/concepts/services-networking/ingress/>`_
+to get traffic into your JupyterHub. This varies wildly
+based on how your cluster was set up, which is why this section is labeled 'Advanced'.
+
+You can enable the required ``ingress`` object with the following in your
+``config.yaml``
+
+.. code-block:: yaml
+
+   ingress:
+       enabled: true
+       hosts:
+        - <hostname>
+
+You can specify multiple hosts that should be routed to the hub by listing them
+under ``ingress.hosts``.
+
+Note that you need to install and configure an
+`Ingress Controller <https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-controllers>`_
+for the ingress object to work. We recommend the
+`nginx ingress <https://github.com/kubernetes/charts/tree/master/stable/nginx-ingress>`_
+controller maintained by the community.
+
+Automatic HTTPS with ``kube-lego`` & Let's Encrypt
```

Comment:
Shouldn't this be in security and not in 'authentication'?

Reply:
yep, had a brain fart